zio-parser/shared/src/main/scala/zio/parser/Parser.scala

```diff
@@ -1439,3 +1439,76 @@ object Parser {
     final case class AllBranchesFailed[Err](left: ParserError[Err], right: ParserError[Err]) extends ParserError[Err]
   }
 }
+
+final case class ParseFailure[+Err](input: String, error: ParserError[Err]) {
```

Comment:
The main change, a wrapper over the input and parser error

Reply:
I would change it's name, let's say to `StringParserError`. That's what it really is, compared to the generic `ParserError` - because we know the input was a string we can provide this extra error pretty printing.

Reply:
Done, renamed the error